app/services/suspend_account_service.rb

```diff
@@ -49,6 +50,10 @@ def purge_profile!
     @account.save!
   end
 
+  def resolve_reports!
+    @account.targeted_reports.in_batches.update_all(action_taken: true)
```

Comment:
I think this should narrow the scope to only update the unresolved reports, rather than all reports. Also, it has an interesting side effect that we don't know who took the action. I think this is the original reason I left it as wip, @Gargron